lessons/operator/01/README.md

```diff
@@ -1,122 +1,121 @@
-This question is answered in our [monitoring as code whitepaper](https://sensu.io/resources/whitepaper/monitoring-as-code-with-sensu) (bold added for emphasis):
+## Monitoring as Code
 
-> Everything as code initiatives as we know them are built on top of **pipelines that enable organizations to codify complex business requirements as repeatable steps**.
-> The CI pipeline (build+test) paved the way for Continuous Delivery (CD).
-> Together as the CI/CD pipeline (build+test, deploy), they provided the foundation for infrastructure as code (build+test, provision+test, deploy).
-> Like IaC, monitoring as code hooks into this CI/CD pipeline, but also requires a dedicated pipeline for ongoing observation of our systems (build+test, provision+test, deploy, observe).
-> We call this missing link the observability pipeline.
->
-> ![](img/everything-as-code.png)
+There is a strong movement towards managing all aspects of the software industry in the same way that we manage application code. 
+Collectively these are referred to as "everything-as-code" methods.
 
-Simply shoving observability data through a generic pipeline where you can cobble together disparate components to filter and route data to various backends is a great first step in an observability pipeline journey – but how to you manage the overall solution?
-And how do you onboard new users to such a bespoke solution – including teams with a multitude of disparate tools spanning multiple generations of technology?
-Sensu's turn-key observability pipeline provides a centralized control plane and end-to-end solution for monitoring as code, which enables organizations to onboard workloads to the observability pipeline in an incremental manner.
+Sensu is designed specifically to enable this kind of workflow for monitoring.
 
-![](img/technology-adoption-lifecycle.jpg)
+### Sensu is an API, Configurations are YAML
 
-_FIGURE 3: the "as code" technology adoption lifecycle._
+The Sensu _backend_ is a service that provides the API and performs event processing.
+All aspects of the observability pipeline can be modified at runtime through the API. 
+Operators can add new components, modify the behavior of filters and handlers, silence noisy streams of data, or add new database sinks *at any time*, without the need to re-deploy.
 
+Components are defined in YAML files, which are pushed to the API to build the pipeline.
+These YAML files can be stored in source-control alongside your application code, and can be tested, versioned, and managed like any other code file.
+This workflow takes the burden off of system operators, by allowing engineers to self-service their monitoring configurations, with security and access controls at every layer.
 
-## Use cases
+To learn more, please check out our [monitoring as code whitepaper](https://sensu.io/resources/whitepaper/monitoring-as-code-with-sensu).
 
-Sensu's observability pipeline
+### Bonsai Asset Repository
```

Comment:
s/Bonsai Asset Repository/Sensu Runtime Assets/

High level comment on this section: Bonsai is to Sensu what the Docker Hub is to Docker. Docker markets the container (which they don't directly make money on) ahead of the Docker Hub (the thing they actually charge for); typically in the order of "Build. Share. Run." I think we want to borrow from this approach as much as possible. 

We hate that we invented our own lowercase-p "packaging" solution for Sensu instead of using something that already existed, but we are proud of the fact that it's a very thin packaging format (tarballs + HTTP + environment variables), and that it "just works" on Linux, Windows, and Mac without installing a runtime (the major reason why we didn't just use Docker containers). So we should really highlight the "runtime asset" feature of Sensu first & foremost, then also highlight the publicly available Bonsai repository. 

Sidebar: Bonsai is named after the original project acronym: "Sensu Asset Index" (SAI, thus BonSAI). 🤓

Reply:
Love the naming context here!

I agree that for Docker, focusing on the container first, and the delivery mechanism second is the right way to structure that. But I also know that this is a huge point of confusion for Docker users (which is why I wrote the blog post ["Where are Docker Images Stored?"](http://blog.thoward37.me/articles/where-are-docker-images-stored/) back in 2013...). 

Similarly it will be confusing for our users to understand how to author, publish, and then consume plugins/assets with Sensu/Bonsai/private asset mirrors. Luckily our system has a lot less moving parts than Docker's does, which makes it a lot easier to describe/comprehend. 

The reason I focused on Bonsai first is a) it's a proper-noun named thing that we will be referring to that doesn't carry any explicit meaning in it's name, so it needs to be introduced by name and explained as early as possible. b) the asset itself isn't the special sauce of monitoring-as-code workflows, but the delivery mechanism *is*. 

By that I mean... monitoring as code depends on the ability to write *configurations* not *code*/binary assets compiled from code. But you still need to write and deliver plugin code. But you can't include that in your yaml file. So you need to provide that code somehow, and need to do that in a way that is real-time, dynamic, and doesn't require a redeploy. The asset delivery mechanism is what enables that, and it's the piece that is essential to making monitoring as code work in a "declarative configuration only" manner. 

So, it makes sense to first focus on the delivery mechanism. Especially Bonsai is very important because it's functioning as the catalog (at the moment) of all the OSS plugins and integrations which are a huge value of Sensu the product, vs monitoring-as-code as a process. 

I do think introing runtime assets before talking about Bonsai might be helpful here, but not strictly necessary, since binary assets are a pretty well known thing, and doesn't really require a lot of unique-to-Sensu explanation. What does require exposition is how Bonsai enables MaC, and the massive collection of OSS plugins available through it. 

I'll see if I can tweak this a bit, but given our timeline for Ill'21 I won't spend too much time trying to improve this section. We can come back to it later, especially once we get sensu-catalog to be more of a thing, and when we write some docs about private asset mirrors. At tht point I think I'd like to make a lesson that is focused on assets in depth.